pkg/solidity-utils/test/WordCodec.test.ts

```diff
@@ -0,0 +1,83 @@
+import { expect } from 'chai';
+
+import { Contract } from 'ethers';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { ZERO_BYTES32, MAX_UINT64 } from '@balancer-labs/v2-helpers/src/constants';
+import TypesConverter from '@balancer-labs/v2-helpers/src/models/types/TypesConverter';
+
+describe('WordCodec', () => {
+  let lib: Contract;
+
+  sharedBeforeEach('deploy lib', async () => {
+    lib = await deploy('MockWordCodec');
+  });
+
+  function getMaxValue(bits: number): number {
+    return 2 ** bits - 1;
+  }
+
+  function getOverMax(bits: number): number {
+    return 2 ** bits;
+  }
+
+  // Can this be made a loop somehow?
```

Comment:
```
  function getMaxValue(bits: number): BigNumber {
    return getOverMax(bits).sub(1);
  }

  function getOverMax(bits: number): BigNumber {
    return BigNumber.from(1).shl(bits);
  }

  const bitLengths = [5, 7, 10, 16, 31, 32, 64];
  for (const bitLength of bitLengths) {
    it(`validates ${bitLength} bit inserts`, async () => {
      const MAX_VALID = getMaxValue(bitLength);
      const insertFunction = `insertUint${bitLength}`;

      expect(await lib[insertFunction](ZERO_BYTES32, MAX_VALID, 0)).to.equal(TypesConverter.toBytes32(MAX_VALID));
      await expect(lib[insertFunction](ZERO_BYTES32, getOverMax(bitLength), 0)).to.be.revertedWith('CODEC_OVERFLOW');
    });
  }
```

This way works and is loopy. It needs it `TypesConverter.toBytes32` to accept a Bignumberish rather than number as well as these changes.

Reply:
Thanks! Was trying to figure out how to do "lib[insertFunction]". Will remember that.